packages/nextjs/src/config/loaders/wrappingLoader.ts

```diff
@@ -277,85 +278,101 @@ async function wrapUserCode(
-    input: SENTRY_WRAPPER_MODULE_NAME,
-
-    plugins: [
-      // We're using a simple custom plugin that virtualizes our wrapper module and the user module, so we don't have to
-      // mess around with file paths and so that we can pass the original user module source map to rollup so that
-      // rollup gives us a bundle with correct source mapping to the original file
-      {
-        name: 'virtualize-sentry-wrapper-modules',
-        resolveId: id => {
-          if (id === SENTRY_WRAPPER_MODULE_NAME || id === WRAPPING_TARGET_MODULE_NAME) {
-            return id;
-          } else {
-            return null;
-          }
-        },
-        load(id) {
-          if (id === SENTRY_WRAPPER_MODULE_NAME) {
-            return wrapperCode;
-          } else if (id === WRAPPING_TARGET_MODULE_NAME) {
-            return {
-              code: userModuleCode,
-              map: userModuleSourceMap, // give rollup acces to original user module source map
-            };
-          } else {
-            return null;
-          }
+  const wrap = (withDefaultExport: boolean): Promise<RollupBuild> =>
+    rollup({
+      input: SENTRY_WRAPPER_MODULE_NAME,
+
+      plugins: [
+        // We're using a simple custom plugin that virtualizes our wrapper module and the user module, so we don't have to
+        // mess around with file paths and so that we can pass the original user module source map to rollup so that
+        // rollup gives us a bundle with correct source mapping to the original file
+        {
+          name: 'virtualize-sentry-wrapper-modules',
+          resolveId: id => {
+            if (id === SENTRY_WRAPPER_MODULE_NAME || id === WRAPPING_TARGET_MODULE_NAME) {
+              return id;
+            } else {
+              return null;
+            }
+          },
+          load(id) {
+            if (id === SENTRY_WRAPPER_MODULE_NAME) {
+              return withDefaultExport ? wrapperCode : wrapperCode.replace('export { default } from', 'export {} from');
+            } else if (id === WRAPPING_TARGET_MODULE_NAME) {
+              return {
+                code: userModuleCode,
+                map: userModuleSourceMap, // give rollup acces to original user module source map
+              };
+            } else {
+              return null;
+            }
+          },
         },
+
+        // People may use `module.exports` in their API routes or page files. Next.js allows that and we also need to
+        // handle that correctly so we let a plugin to take care of bundling cjs exports for us.
+        commonjs({
+          sourceMap: true,
+          strictRequires: true, // Don't hoist require statements that users may define
+          ignoreDynamicRequires: true, // Don't break dynamic requires and things like Webpack's `require.context`
+          ignore() {
+            // We want basically only want to use this plugin for handling the case where users export their handlers with module.exports.
```

Comment:
q: So ignore is only used to disable the `require` -> `import` rewriting? But we need other functionality from commonjs?

Reply:
This plugin + option have been in this file before. But you are correct. We're only using this plugin to follow `module.exports`s of the modules we're actually wrapping, but we do not want to turn `import`s into `require`s in our wrapper. This plugin does both.

Reply:
Yeah, I was just curious, thx for clarifying